sdk/servicebus/service-bus/review/service-bus.api.md

```diff
@@ -304,75 +228,14 @@ export interface SessionReceiverOptions {
     sessionId: string | undefined;
 }
 
-// @public
-export type SqlAction = SqlFilter;
-
-// @public
-export interface SqlFilter {
-    compatibilityLevel?: number;
-    requiresPreprocessing?: boolean;
-    sqlExpression?: string;
-    sqlParameters?: SqlParameter[];
-}
-
-// @public
-export type SqlParameter = {
-    key: string;
-    value: string | number;
-    type: string;
-};
-
 // @public
 export interface SubscribeOptions extends OperationOptions, MessageHandlerOptions {
 }
 
 // @public
-export interface SubscriptionDetails {
-    accessedOn?: string;
-    autoDeleteOnIdle: string;
-    createdOn: string;
-    deadLetteringOnFilterEvaluationExceptions: boolean;
-    deadLetteringOnMessageExpiration: boolean;
-    defaultMessageTtl?: string;
-    defaultRuleDescription?: any;
-    enableBatchedOperations: boolean;
-    enablePartitioning?: boolean;
-    entityAvailabilityStatus: string;
-    forwardDeadLetteredMessagesTo?: string;
-    forwardTo?: string;
-    lockDuration: string;
-    maxDeliveryCount: number;
-    maxSizeInMegabytes?: number;
-    messageCount: number;
-    messageCountDetails?: MessageCountDetails;
-    requiresSession: boolean;
-    sizeInBytes?: number;
-    status?: EntityStatus;
-    subscriptionName: string;
-    topicName: string;
-    updatedOn: string;
-    userMetadata?: string;
-}
-
-// @public
-export interface SubscriptionOptions {
-    autoDeleteOnIdle?: string;
-    deadLetteringOnFilterEvaluationExceptions?: boolean;
-    deadLetteringOnMessageExpiration?: boolean;
-    defaultMessageTtl?: string;
-    enableBatchedOperations?: boolean;
-    forwardDeadLetteredMessagesTo?: string;
-    forwardTo?: string;
-    lockDuration?: string;
-    maxDeliveryCount?: number;
-    requiresSession?: boolean;
-    status?: EntityStatus;
-    userMetadata?: string;
-}
-
-// @public
-export interface SubscriptionRuleManagement {
+export interface SubscriptionRuleManager {
     addRule(ruleName: string, filter: boolean | string | CorrelationFilter, sqlRuleActionExpression?: string): Promise<void>;
+    close(): Promise<void>;
```

Comment:
Why do we need close() on the subscription rule manager?
Close on the client, sender and receiver actually closes the AMQP connection, AMQP sender link and AMQP receiver link respectively.
What would the close on the rule manager close?

The $management link used by the rule manager would ideally be shared by the sender and the receiver if any

Reply:
It will close the management client link if it's the last usage of it the management link.

It runs through all the same logic as any other client connection context.